pkg/api/v1/atlasdatabaseuser_types.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2020 MongoDB.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+)
+
+// Important:
+// The procedure working with this file:
+// 1. Edit the file
+// 1. Run "make generate" to regenerate code
+// 2. Run "make manifests" to regenerate the CRD
+
+func init() {
+	SchemeBuilder.Register(&AtlasProject{}, &AtlasProjectList{})
+}
+
+type ScopeType string
+
+const (
+	ClusterScopeType  ScopeType = "CLUSTER"
+	DataLakeScopeType ScopeType = "DATA_LAKE"
+)
+
+// AtlasDatabaseUserSpec defines the desired state of Database User in Atlas
+type AtlasDatabaseUserSpec struct {
+	// Project is a reference to AtlasProject resource the user belongs to
+	Project ResourceRef `json:"projectRef"`
+
+	// DatabaseName is a Database against which Atlas authenticates the user. Must be equal to 'admin'
+	// as only SCRAM authentication is supported so far.
+	DatabaseName string `json:"databaseName"`
+
+	// DeleteAfterDate is a timestamp in ISO 8601 date and time format in UTC after which Atlas deletes the user.
+	// The specified date must be in the future and within one week of the time you make the API request.
+	DeleteAfterDate string `json:"deleteAfterDate,omitempty"`
+
+	// Labels is an array containing key-value pairs that tag and categorize the database user.
+	// Each key and value has a maximum length of 255 characters.
+	Labels []LabelSpec `json:"labels,omitempty"`
+
+	// Roles is an array of this user's roles and the databases / collections on which the roles apply. A role allows
+	// the user to perform particular actions on the specified database.
+	Roles []RoleSpec `json:"roles"`
+
+	// Scopes is an array of clusters and Atlas Data Lakes that this user has access to.
+	Scopes []ScopeSpec `json:"scopes"`
```

Comment:
Scopes should be `omitempty` to support project-wide users

Reply:
I think in the scope document there were suggestions to make this mandatory (I believe from Jason Mimick): https://docs.google.com/document/d/1hZrcQy18Calv5hwkxvcgHO_AHKusy6-O2by7uaETdi8/edit#bookmark=id.4q1yljo2rfvh

I'm not sure what is the best approach here, I think we need the second opinion on this, @jamesbroadhead @theburi what do you think?

Reply:
I'll defer to @theburi. It seems to me that we should support project-wide users

Reply:
^ and that not requiring people to fill this field would be good for demo use-cases

Reply:
Have just checked - so having the `scopes` a mandatory still should allow to get project-wide users by just specifying `scopes:[]`

So this will be an intentional configuration done by a user, the positive sign I see is that they won't dismiss this just because they are not aware of such field.
WDYT?

Reply:
probably nice if they could leave it out & for the operator to use '[]'